genesyscloud/resource_cache/resource_cache.go

```diff
@@ -25,6 +26,12 @@ func SetCache[T any](cache CacheInterface[T], key string, value T) {
 	}
 }
 
+func DeleteCacheItem[T any](cache CacheInterface[T], key string) {
```

Comment:
I am not sure why we have a DeleteCache Item method.  What is the use case you are running into?  Just trying to understand because generally the cache is used for the exporter and I am not sure I see the use case for the delete.

Reply:
In my acceptance tests, I created an export resource and a campaign. The `deleteOutboundCampaign` method was getting stuck because it deletes the campaign and then reads by ID with retries until it receives a 404. Because the export resource is also present, `IsExporterActive` returns true, and the campaign is read from cache. That causes the test to fail and claim that the campaign still exists, even though it does not. 

This can happen with any resource that does reads to confirm that the object has been deleted before exiting the deleteContext function.

I haven't been able to recreate this behaviour using the locally compiled binary, so I'm not sure yet what the difference is with test cases that allows this to happen.